From 25f982f3469fb34307c4744852ba5de0c20e2546 Mon Sep 17 00:00:00 2001
From: Jeegar Patel <jeegar.patel@xilinx.com>
Date: Fri, 25 Sep 2020 02:11:18 -0700
Subject: [PATCH 1676/1852] drm: xlnx: zynqmp_dp: Fix CHECKED_RETURN type
 warnings

commit dc02c6c791f6ea39059daca09f8c85840355404b from
https://github.com/Xilinx/linux-xlnx.git

This patch fix CHECKED_RETURN type coverity warnings.

Calling "drm_dp_dpcd_writeb" without checking return value.
Calling "pm_runtime_get_sync" without checking return value.

Signed-off-by: Jeegar Patel <jeegar.patel@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_dp.c | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_dp.c b/drivers/gpu/drm/xlnx/zynqmp_dp.c
index e9a95cb94a3b..debb287904fc 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_dp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_dp.c
@@ -1530,7 +1530,12 @@ static void zynqmp_dp_encoder_enable(struct drm_encoder *encoder)
 	unsigned int i;
 	int ret = 0;
 
-	pm_runtime_get_sync(dp->dev);
+	ret = pm_runtime_get_sync(dp->dev);
+	if (ret < 0) {
+		dev_err(dp->dev, "IRQ sync failed to resume: %d\n", ret);
+		return;
+	}
+
 	dp->enabled = true;
 	zynqmp_dp_init_aux(dp);
 	zynqmp_dp_update_misc(dp);
@@ -1561,11 +1566,17 @@ static void zynqmp_dp_encoder_disable(struct drm_encoder *encoder)
 {
 	struct zynqmp_dp *dp = encoder_to_dp(encoder);
 	void __iomem *iomem = dp->iomem;
+	int ret;
 
 	dp->enabled = false;
 	cancel_delayed_work(&dp->hpd_work);
 	zynqmp_dp_write(iomem, ZYNQMP_DP_TX_ENABLE_MAIN_STREAM, 0);
-	drm_dp_dpcd_writeb(&dp->aux, DP_SET_POWER, DP_SET_POWER_D3);
+	ret = drm_dp_dpcd_writeb(&dp->aux, DP_SET_POWER, DP_SET_POWER_D3);
+	if (ret < 0) {
+		dev_err(dp->dev, "failed to write a byte to the DPCD: %d\n",
+			ret);
+		return;
+	}
 	zynqmp_dp_write(iomem, ZYNQMP_DP_TX_PHY_POWER_DOWN,
 			ZYNQMP_DP_TX_PHY_POWER_DOWN_ALL);
 	if (zynqmp_disp_aud_enabled(dp->dpsub->disp))
-- 
2.31.1

